bindings_generator/src/lib.rs

```diff
@@ -149,25 +129,20 @@ fn generate_class_bindings(
 
         generate_free_impl(output_trait_impls, &api, class)?;
 
-        if !class.base_class.is_empty() {
-            generate_deref_impl(output_trait_impls, class)?;
-        }
+        generate_deref_impl(output_trait_impls, class)?;
 
-        if class.is_refcounted() {
-            generate_impl_ref_counted(output_trait_impls, class)?;
-            generate_reference_clone(output_trait_impls, class)?;
-            generate_drop(output_trait_impls, class)?;
-        }
+        // RefCounted
+        generate_impl_ref_counted(output_trait_impls, class)?;
+        generate_reference_clone(output_trait_impls, class)?;
+        generate_drop(output_trait_impls, class)?;
```

Comment:
I think the old way of having a branch here communicates the intention better than a comment.